packages/@aws-cdk/aws-stepfunctions-tasks/lib/lambda/invoke.ts

```diff
@@ -46,6 +46,15 @@ export interface LambdaInvokeProps extends sfn.TaskStateBaseProps {
    * @default - Version or alias inherent to the `lambdaFunction` object.
    */
   readonly qualifier?: string;
+
+  /**
+   * Whether the Step Functions resource is Lambda:invoke or the function ARN.
+   *
+   * ARN is not compatible with the WAIT_FOR_TASK_TOKEN IntegrationPattern.
```

Comment:
It'll be more helpful for users if we detect this condition and throw an error if they have supplied both an integration pattern and a request to use the function ARN

we should validate this condition and warn users.